### PR TITLE
Feat/add import export data

### LINKED ISSUE #411: [Feature] Add import/export to data management
**Problem**
There is no way to import data into TakeNote.

**Solution**
In Data Management, add import and export options. Notes and categories should export as a single `.json` file, and you should be able to import that back into the app.

takenote-data.json:

```json
{
  "notes": [],
  "categories": []
}
```

That way, your entire database can be backed up into a single file, and imported into another instance of the app.

**Notes**
I don't think settings is necessary here.

### PR DESCRIPTION
**Pull request by JoseRFelix**
## Description

Allow users to backup and import their Takenote data. The data is exported as JSON and imported likewise. 

Edge cases handled when importing: 

- If duplicate categories are found then they are overwritten and won't be added again. This is to avoid key collisions. 
- Duplicate notes are added with new Ids.
- Scratchpad is overwritten (Don't know if it would be better to merge with the current user's content). 

Closes #411

### Browser checklist

This PR has been tested in the following browsers:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### Testing checklist

- [x] End-to-end tests have been created if necessary

### CODE REVIEW

File: src/client/utils/helpers.ts [hunk 1]
```diff
@@ -88,6 +88,27 @@ export const downloadNotes = (notes: NoteItem[], categories: CategoryItem[]): vo
...
+
+  const date = new Date()
+  const backupDate = `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`
+
+  const json = JSON.stringify({ notes, categories })
+  const blob = new Blob([json], { type: 'application/json' })
+
+  const downloadUrl = window.URL.createObjectURL(blob)
+  pom.href = downloadUrl
+  pom.download = `takenote-backup-${new Date().toISOString()}.json`
```
Comment (taniarascia) [inline src/client/utils/helpers.ts:102]: Can probably use `dayjs` to make a slightly more pleasant date URL.

File: src/resources/LabelText.ts [hunk 2]
```diff
@@ -23,5 +23,7 @@ export enum LabelText {
   RENAME = 'Rename category',
   ADD_CONTENT_NOTE = 'Please add content to this new note to access the menu options.',
   DOWNLOAD_ALL_NOTES = 'Download all notes',
+  BACKUP_ALL_NOTES = 'Backup all notes',
+  IMPORT_BACKUP = 'Import Backup',
```
Comment (taniarascia) [inline src/resources/LabelText.ts:27]: Slight nitpick, `Import backup` with lowercase `b` to match the case of the rest of the strings.

**Review by taniarascia — CHANGES_REQUESTED** [review]

**Comment by taniarascia** [comment]
Awesome PR! Thanks a lot! Sorry, another one went in that made a slight conflict, so just made a few notes, then it's good to go. Also, I think it should run a `sync` right after the upload since it's left in a pending state.

**Comment by JoseRFelix** [comment]
> Awesome PR! Thanks a lot! Sorry, another one went in that made a slight conflict, so just made a few notes, then it's good to go. Also, I think it should run a `sync` right after the upload since it's left in a pending state.

Great! Thanks for the review, I have made the changes accordingly.

**Review by taniarascia — APPROVED** [review]

**Pull request merged**